### ISSUE TITLE
Else if statements and then statements error when removed

### ISSUE THREAD

**Issue by lamqson** [dsherret/ts-morph#708]
**Describe the bug**

Version: 4.0.1
React: 16.10.0
MUI: 4.4.3

**To Reproduce**

```ts
import { Project, ts } from "ts-morph";

console.log(ts.version); // x.x.x <-- provide this
source transform @material-ui/core/esm/FormLabel/FormLabel.js
  source transform @material-ui/core/esm/MenuItem/index.js
  source transform @material-ui/core/esm/MenuItem/MenuItem.js
  source transform @material-ui/core/esm/ListItem/index.js
  source transform @material-ui/core/esm/ListItem/ListItem.js
(node:48180) UnhandledPromiseRejectionWarning: Error: Error replacing tree: Should not have children left over.
-- Details --
Path: /MyClass.tsx
Text: "... if (autoFocus) {\n      if (listItemRef.current) {\n        listItemRef.current.focus();\n      } else\n    }\n  }, [autoFocus]);\n  var children = React.Children.toArray(childrenProp);\n  var hasSecondaryA..."
Stack: Error: Error replacing tree: Should not have children left over.
    at DefaultParentHandler.handleNode (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5734:19)
    at ParentFinderReplacementNodeHandler.handleNode (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5799:36)
    at NodeHandlerHelper.handleForValues (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5611:21)
    at ParentFinderReplacementNodeHandler.handleChildren (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5673:25)
    at ParentFinderReplacementNodeHandler.handleNode (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5663:18)
    at ParentFinderReplacementNodeHandler.handleNode (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5802:19)
    at NodeHandlerHelper.handleForValues (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5611:21)
    at ParentFinderReplacementNodeHandler.handleChildren (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5673:25)
    at ParentFinderReplacementNodeHandler.handleNode (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5663:18)
    at ParentFinderReplacementNodeHandler.handleNode (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:5802:19)
    at doManipulation (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:6460:15)
    at removeChildrenWithFormatting (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:6730:5)
    at removeStatementedNodeChildren (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:6806:5)
    at removeStatementedNodeChild (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:6803:5)
    at IfStatement.remove (/Users/sonlam/workspaces/octupi/raindrop/node_modules/ts-morph/dist/ts-morph.js:13181:9)
    at setResult (/Users/sonlam/workspaces/octupi/raindrop/node_modules/fuse-box/production/transformation/conditionUnwrapper.js:63:18)
    at onIfStatement (/Users/sonlam/workspaces/octupi/raindrop/node_modules/fuse-box/production/transformation/conditionUnwrapper.js:77:17)
    at /Users/sonlam/workspaces/octupi/raindrop/node_modules/fuse-box/production/transformation/conditionUnwrapper.js:107:13
    at Array.forEach (<anonymous>)
    at Object.conditionUnwrapperProduction (/Users/sonlam/workspaces/octupi/raindrop/node_modules/fuse-box/production/transformation/conditionUnwrapper.js:105:17)
(node:48180) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:48180) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
const project = new Project();
const sourceFile = project.createSourceFile("test.ts", ``);

// reproduce the problem here
```

**Expected behavior**
TS compilation should not fail here

**Comment by dsherret** [comment]
@lamqson Could you provide some reproduction steps? Thanks!

**Comment by dsherret** [comment]
@lamqson can you upgrade to 4.1.0 and re-run the code? After you upgrade, it should tell you about a syntax error in the code being created. Let me know if it helps you solve the problem. If not, then I have to improve that error message more somehow.

**Comment by jcmoore0** [comment]
@dsherret @lamqson  - Here is a repository reproducing the error:  https://github.com/jcmoore0/Fuse-Ts-Morph

These 2 libraries both fail when making a production bundle (yarn start dist)

@material-ui/core: 4.4.3
react-final-form: 6.3.0

Thanks!

**Comment by dsherret** [comment]
This is a fusebox issue. A syntax error is being inserted:

```ts
else
    }
```

I will comment on that issue.

**Comment by dsherret** [comment]
Oh, seems like the fusebox issue referenced is a different issue. Will look into it!

**Comment by dsherret** [comment]
I've added two failing tests for this. Will look into it later though.

**Comment by dsherret** [comment]
Will be fixed in 4.1.1.
